packages/ember-glimmer/tests/integration/syntax/with-test.js

```diff
@@ -30,4 +38,156 @@ moduleFor('Syntax test: {{#with}}', class extends SharedConditionalsTest {
 
     this.assertText('False');
   }
+
+  ['@test updating the context should update the alias']() {
+    this.render(`{{#with cond1 as |cond|}}{{cond.greeting}}{{else}}False{{/with}}`, {
+      cond1: { greeting: 'Hello' }
+    });
+
+    this.assertText('Hello');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Hello');
+
+    this.runTask(() => set(this.context, 'cond1', { greeting: 'Hola'}));
+
+    this.assertText('Hola');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Hola');
+
+  }
+
+  ['@test the scoped variable is not available outside the {{with}} block.']() {
+    this.render(`{{name}}-{{#with other as |name|}}{{name}}{{/with}}-{{name}}`, {
+      name: 'Stef',
+      other: 'Yehuda'
+    });
+
+    this.assertText('Stef-Yehuda-Stef');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Stef-Yehuda-Stef');
```

Comment:
@chancancode I think we're both confused, completely agree with the pattern you described. I should just asked for clarification in terms of the actual test plan pattern. What I mean by this is that the `SharedConditionalTests` have the `templateFor`/`generator` pattern. I thought you were asking to follow that setup for each `moduleFor` in this file.